src/ParametricOptInterface.jl

```diff
@@ -818,25 +843,100 @@ function MOI.optimize!(model::ParametricOptimizer)
         end
         empty!(model.updated_parameters)
     end
+
     MOI.optimize!(model.optimizer)
+
+    if model.evaluate_duals
+        calculate_dual_of_parameters(model)
+    end
 end
 
 ### Duals
 
-function MOI.get(model::ParametricOptimizer, attr::MOI.ConstraintDual, ci::MOI.ConstraintIndex)
-    return MOI.get(model.optimizer, attr, ci)
+function calculate_dual_of_parameters(model::ParametricOptimizer) 
+    param_dual_cum_sum = Dict{Int, Float64}()
+
+    for vi in keys(model.parameters)
+        param_dual_cum_sum[vi.value] = 0.0
+    end
+    
+    # TODO
+    # Quadratic constraints
+    for S in ScalarLinearSetList
+        affine_constraint_cache_inner = model.affine_constraint_cache[MOI.ScalarAffineFunction{Float64}, S{Float64}]
+
+        if length(affine_constraint_cache_inner) > 0
+            param_dual_cum_sum = iterate_over_constraint_cache_affine(model, affine_constraint_cache_inner ,param_dual_cum_sum)
+        end
+
+        # for F in [MOI.ScalarQuadraticFunction]
+        #     affine_constraint_cache_inner = model.affine_constraint_cache[F{Float64}, S{Float64}]
+
+        #     if length(affine_constraint_cache_inner) > 0
+        #         param_dual_cum_sum = iterate_over_constraint_cache_quadratic(model, affine_constraint_cache_inner ,param_dual_cum_sum)
+        #     end
+        # end
+    end
+
+    for param in model.affine_objective_cache
+        vi = param.variable_index.value
+        param_dual_cum_sum[vi] += param.coefficient
+    end
+
+    empty!(model.dual_value_of_parameters)
+    for (vi_val, param_dual) in param_dual_cum_sum
+        model.dual_value_of_parameters[MOI.VariableIndex(vi_val)] = param_dual
+    end
+end
+
+function iterate_over_constraint_cache_affine(model::POI.ParametricOptimizer, constraint_cache_inner::DD.WithType{F, S}, param_dual_cum_sum::Dict{Int, Float64}) where {F,S}
+    for (ci, param_array) in constraint_cache_inner
+        param_dual_cum_sum = calculate_parameters_in_ci(model.optimizer, param_array, ci, param_dual_cum_sum)
+    end
+
+    return param_dual_cum_sum
+end
+
+# TODO
+# Quadratic constraints
+function iterate_over_constraint_cache_quadratic(model::POI.ParametricOptimizer, constraint_cache_inner::DD.WithType{F, S}, param_dual_cum_sum::Dict{Int, Float64}) where {F,S}
+    for (poi_ci, param_array) in model.quadratic_constraint_cache_pc
+        moi_ci = model.quadratic_added_cache[poi_ci]
+        for param in param_array
+            if cp.value == param.variable_index.value
+                cons_dual = MOI.get(model.optimizer, MOI.ConstraintDual(), moi_ci)
+                param_dual_quadratic_constraint_affine_part += cons_dual*param.coefficient
+            end
+        end
+    end
+    
+    for (ci, param_array) in constraint_cache_inner
+        param_dual_cum_sum = calculate_parameters_in_ci(model.optimizer, param_array, ci, param_dual_cum_sum)
+    end
```

Comment:
this does not make much sense. Is it correct?

Reply:
Nope, I was going to comment that out